update-testsuite.py

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env python3
+
+# Script to automatically update all tests in this repository based on the
+# status of the upstream repositories themselves. This will prepare a git
+# submodule-of-sorts (not literally) in `repos/spec` with all the upstream repos
+# fetched into that one location. Tests will be diff'd against the merged
+# version of the upstream spec repo and the proposal repo, and if there's a
+# difference then the test is included.
+
+import subprocess
+import os
+import shutil
+
+def main():
+    spec = Repo('spec')
+    spec3 = Repo('spec', branch='wasm-3.0')
+
+    repos = []
+    repos.append(spec3)
+    repos.append(Repo('threads'))
+    repos.append(Repo('exception-handling'))
+    repos.append(Repo('gc'))
+    repos.append(Repo('tail-call'))
+    repos.append(Repo('annotations'))
+    repos.append(Repo('function-references'))
+    repos.append(Repo('memory64', upstream=spec3))
+    repos.append(Repo('extended-const'))
+    repos.append(Repo('multi-memory'))
+    repos.append(Repo('relaxed-simd'))
+    repos.append(Repo('custom-page-sizes'))
+    repos.append(Repo('wide-arithmetic'))
+
+    # Make sure that `repos/spec` is a git repository
+    if not os.path.isdir('repos/spec'):
+        git('init', 'repos/spec')
```

Comment:
If these is just one repo these days maybe we don't need two levels of nesting here?